nixup/modules/config/resourceFiles.nix

```diff
@@ -0,0 +1,90 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  options = {
+    user.resourceFiles = mkOption rec {
+      default = {};
+      type = types.loaOf (types.submodule options);
+      example = literalExample ''
+        [ { target = ".ssh/config";
```

Comment:
could this be simplified to an attribute set where the key is the file name and the value is the string contents? Nix already has utilities for e.g. reading in file contents as a string (bulitins.readFile), so it seems unecessary to reinvent that with the "source" option